### ISSUE TITLE
Default branch name when creating branch from remote

### ISSUE THREAD

**Issue by ElfSundae** [ForkIssues/Tracker#119]
![image](https://user-images.githubusercontent.com/526008/31352090-0e24e450-acf3-11e7-8ca7-4385caca4c89.png)

When creating a new branch from remote, please provider a default branch name, and select it in text input just like "Rename Remote" does. Thanks.

![image](https://user-images.githubusercontent.com/526008/31352180-65518102-acf3-11e7-90fd-637519c62636.png)

**Comment by DanPristupov** [comment]
@ElfSundae I'm not sure I understand the point.
Why do you need to start a local branch from remote? I can't imagine a reason to do this in real life. 

May be you want to create a tracking reference for the remote branch? If so, you just need to checkout remote branch and it will propose to track remote branch (i.e. create a corresponding local branch which tracks the remote one).

Could you please explain your use case?

**Comment by ElfSundae** [comment]
Yes, I just want to checkout a remote branch to local.

Then what is the difference between "Checkout" and "Create New Branch" in the remote branch popup menu?

**Comment by clounie** [comment]
Create new branch lets you create a new branch, that has no relation to anything on the remote. 

It's a convenience thing, I think - and personally the only way I create new branches in Fork.

**Comment by DanPristupov** [comment]
@ElfSundae 
The relation between local and remote branch is called **tracking**. Git uses this relation for Push/Pull. More information here: https://git-scm.com/book/en/v2/Git-Branching-Remote-Branches

**Create New Branch**. When you create a local branch, it's just a local branch. No relations to any other branch.

**Checkout**. Technically there's no way to checkout a remote branch. You can only checkout a local one. So basically when you checkout remote branch Fork does the following:
- if remote branch does have a corresponding local branch, Fork will checkout it.
- if remote branch doesn't have a corresponding local branch which tracks it, it will create one and then checkout.

**Comment by ElfSundae** [comment]
@DanPristupov @clounie Thanks very much for your so kind explanation of Git basic. Sorry I did not express the issue clearly and technically. I am not good at English.

I am just confused in the **Popup menu of Remotes branch**:

![image](https://user-images.githubusercontent.com/526008/31559773-7d13af50-b017-11e7-9a44-505b89bdff78.png)
![image](https://user-images.githubusercontent.com/526008/31560163-e5676e2e-b018-11e7-91ba-9449d318956b.png)

It seems "Create new branch at remote" means creating a new branch and tracking it with this remote branch, the Git command would be `git branch 5.0 upstream/5.0`, am I right? 

I cloned a repo which has many branches, I wanted to create some local branches and track them to corresponding remote branches, without checkout, so I clicked this "Create New Branch" menu item under Remotes. For each creating, I need to input a branch name, so I opened this issue to suggest giving a default name. 
Maybe I misunderstand something, and this suggestion is very slight, so I will just close this issue.
